lib/client.ts

```diff
@@ -1,13 +1,303 @@
 import { datasets } from './datasets';
 import { users } from './users';
-import { ClientOptions } from './httpClient';
+import HTTPClient, { ClientOptions } from './httpClient';
+import { gzip } from 'zlib';
+import { promisify } from 'util';
+import { Readable, Stream } from 'stream';
 
-export default class Client {
+export default class Client extends HTTPClient {
     datasets: datasets.Service;
     users: users.Service;
+    localPath = '/api/v1/datasets';
```

Comment:
Should this be `/api/v1` because it's not a specific client?

Reply:
updated